### PR TITLE
travis: do a "cargo check" (no codegen) for all crates

### PR DESCRIPTION
**Pull request by birkenfeld**
This shows a failure in stm32l100 that seems legit...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+language: rust
+
+rust:
+  - stable
+
 install:
-  - pip install pyyaml
+  - wget https://bootstrap.pypa.io/get-pip.py
+  - python3 get-pip.py --user
+  - python3 -m pip install --user pyyaml
```
Comment (adamgreig) [inline .travis.yml]: `python3-yaml` is on the Travis apt whitelist, so maybe we could replace this with
```
addons:
  apt:
    packages:
      - python3
```

File: Makefile [hunk 2]
```diff
@@ -46,6 +50,10 @@ $(1)/src/%/.form: $(1)/src/%/mod.rs
 	mv $$$$(dirname $$<)/lib.rs $$<
 	find $$$$(dirname $$<) -name "*.rs" -exec rustfmt {} +
 	touch $$$$(dirname $$<)/.form
+
+$(1)/src/%/.check: $(1)/src/%/mod.rs
+	cd $(1) && cargo check --target-dir ../target/check/$$* --features $$*
+
```
Comment (adamgreig) [inline Makefile]: You should also touch `.check` so make sees it get created.

**Comment by adamgreig** [comment]
I'm tracking down the build failure, by the way. Half resolved.

**Comment by adamgreig** [comment]
If you rebase onto master the build will hopefully succeed.

File: .travis.yml (hunk 1 above)
Comment (birkenfeld) [reply .travis.yml]: Nice!

**Comment by birkenfeld** [comment]
Passing :fireworks: The build takes about 30 minutes on Travis, which I guess is fair enough, it has to build 46 crates.

Each build uses its own target directory since otherwise the 2 cores offered by Travis can't straightforwardly be used in parallel, since the same crate is built multiple times leading to interlocks. On the other hand, the artifacts that could be shared only take a few seconds to compile.

I also tried building each crate in one go with all supported features, but it led to OOM and timeouts.

**Comment by adamgreig** [comment]
Looks good, thanks!

**Pull request merged**